File: TS/fib.ts
import { performance } from "perf_hooks";

function fib(num: number): number {
    if (!num)return 0;
    let prev: number = 0;
    let now: number = 1;
    for (let i = 1;i < num;++i) {
        let x = prev;
        prev = now;
        now += x;

    }
    return now;
}

console.log(fib(2));

File: C++/fib.cc
#include <iostream>

#define ll long long

ll fib(int num) {
    if (!num)return 0;
    ll prev = 0;
    ll now = 1;
    for (int i = 1;i < num;++i) {
        int x = prev;
        prev = now;
        now = x + now;
    }
    return now;
}

int main(void) {
    for (int i = 0;i < 100;++i) {
        std::cout << fib(i) << std::endl;
    }
    return 0;
}